docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,190 @@
+keep EmailAlertApi's database manageable going forward. There are a couple of
+different approaches to consider, including storing unused data in cloud
+storage away from the systems database (like we do when archiving emails) and
+subsequently deleting the data locally. Or, setting up a means of removing the
+data we don't use from the system entirely.
+
+[archived]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_archive_worker.rb
+[deleted]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_deletion_worker.rb
+[initial policy]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-data-retention.md
+[migrate subscribers to daily digest emails]: https://github.com/alphagov/email-alert-api/pull/1352
+
+## Decision
+
+We decided that historic, unused data should be removed periodically from the
+system entirely. It was concluded that the data the system no longer uses
+should only exist for a maximum of one year before removal. We sometimes
+analyse historic data from up to a year ago but we have no evidence of needing
+this data for analytics beyond one year. There is however some bespoke
+behaviour for subscriber lists where we delete certain lists sooner than a year
+but that will be elaborated on in a later section.
+
+We chose to remove the historic data entirely instead of retaining it through
+archiving as the latter option seems unnecessary. If we've deemed the data as
+historic and don't see a use for it (analytical or otherwise) going forward,
+then retaining it via an external data store seems a waste of resources and
+energy in maintaining the process.
+
+It was also concluded that we don't need to change the retention for emails
+(and subscription contents) as we already clean them up via the [email deletion
+worker][deleted] which is sufficient.
+
+The following list documents our approach to the decision for different data
+in the system.
+
+* **ContentChange, Message & DigestRun**. Remove once their `created_at` value is
+  over a year old.
+
+* **MatchedContentChange, MatchedMessage & DigestRunSubscriber**. Cascaded when
+  their parent content change / message / digest run is deleted.
+
+* **Subscription**. Remove subscriptions when they've been ended for over a
+  year, based on their `ended_at` value. Once a subscription is ended it [can't
+  currently be made active again] and so can be deleted.
+
+* **Previously subscribed to SubscriberList**. Remove lists when they have no
+  active subscriptions and their latest, ended subscription is over a year old.
+  It is possible for a list we deem as historic to be subscribed to again,
+  however this seems unlikely due to the list not having any active
+  subscriptions for over a year.  If this was to happen, after the list has
+  been deleted then a new `SubscriberList` record will be generated.
+
+* **Unused SubscriberList**. Remove lists which have never had any
```

Comment:
Minor: we could make this a separate paragraph, since it's different to the other items in the list, which are all about retaining data for 1 year.

Reply:
Think this is a duplicate of a previous comment, I've responded to that one so resolving this one! 🤔